docs/tutorials/generate-custom-test-data.rst

```diff
@@ -0,0 +1,120 @@
+.. _ksql_generate-custom-test-data:
+
+Generate custom test data by using the ksql-datagen tool
+========================================================
+
+KSQL enables generating test data that complies with a custom schema
+that you define. 
+
+To generate test data, create an Apache Avro schema and pass it to the
+command-line tool, ``ksql-datagen``, which generates random data according
+to the schema you provide.
+
+**Prerequisites:** 
+
+- :ref:`Confluent Platform <installation>` is installed and running.
+  This installation includes a Kafka broker, KSQL, |c3-short|, |zk|,
+  Schema Registry, REST Proxy, and Kafka Connect.
+- If you installed |cp| via TAR or ZIP, navigate to the installation
+  directory. The paths and commands used throughout this tutorial assume
+  that you're in this installation directory.
+- Java: Minimum version 1.8. Install Oracle Java JRE or JDK >= 1.8 on your
+  local machine.
+
+Start a Kafka cluster, including |zk|, a Kafka broker, and |sr|. KSQL queries
+messages from this Kafka cluster. KSQL is installed in the |cp| by default.
+
+Define a custom schema
+----------------------
+
+In this example, you register a custom Avro schema to the |sr|. The schema is
+named `impressions.avro 
+<https://github.com/apurvam/streams-prototyping/blob/master/src/main/resources/impressions.avro>`_, 
+and it represents ads delivered to users.
+
+#. Download impressions.avro and copy it to a convenient directory. It's used
+   by ``ksql-datagen`` when you start generating test data.
+
+   .. code:: bash
+
+      curl https://raw.githubusercontent.com/apurvam/streams-prototyping/master/src/main/resources/impressions.avro > impressions.avro
+
+#. Use the ``curl`` command to post the ``impressions-key`` schema to the |sr|.
+
+   .. code:: bash
+
+      curl -X POST -H "Content-Type: application/vnd.schemaregistry.v1+json" \
+      --data '{"schema": "{\"namespace\":\"streams\",\"name\":\"impressions\",\"type\":\"string\"}"}' \
+      http://localhost:8081/subjects/impressions-key/versions
+
+   Your output should resemble:
+
+   .. code:: bash
+
+      {"id":1}
+
+# Use the ``curl`` command to post the ``impressions-value`` schema to the |sr|.

```

Comment:
missing a period after the hash `#.` causes this:

![image](https://user-images.githubusercontent.com/11722533/41300427-8fcf6910-6e1a-11e8-9056-a982085a1160.png)